ci: Update CodeQL Action from v2 to v3

## Problem

GitHub deprecated CodeQL Action v1 and v2 as of January 10, 2025. The `gosec` security scan step fails with:

```
##[error]CodeQL Action major versions v1 and v2 have been deprecated.
Please update all occurrences of the CodeQL Action in your workflow files to v3.
```

## Current Configuration

In `.github/workflows/test.yml`:
```yaml
- name: Upload SARIF file
  uses: github/codeql-action/upload-sarif@v2  # DEPRECATED
```

## Fix

Update to v3:
```yaml
- name: Upload SARIF file
  uses: github/codeql-action/upload-sarif@v3
```

## Impact

- CI checks show `gosec` as failing even though security scan passes
- PRs require admin override to merge
- Blocking normal CI flow

## References

- [GitHub Changelog: CodeQL Action v2 Deprecated](https://github.blog/changelog/2025-01-10-code-scanning-codeql-action-v2-is-now-deprecated/)

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -181,6 +181,6 @@ jobs:
         args: '-no-fail -fmt sarif -out results.sarif ./...'
 
     - name: Upload SARIF file
-      uses: github/codeql-action/upload-sarif@v2
+      uses: github/codeql-action/upload-sarif@v3
       with:
         sarif_file: results.sarif

